core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -3119,12 +3110,9 @@ struct ViewValueFunctor<DeviceType, ValueType, true /* is_scalar */> {
       if (Kokkos::Profiling::profileLibraryLoaded()) {
         Kokkos::Profiling::endParallelFor(kpID);
       }
-    }
-#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
-    else {
+    } else {
```

Comment:
and here.

Reply:
The `else` branch should be removed just as above (and the brace in line 3112/3107 restored).